src/NuGet.Core/NuGet.Common/NuGetEnvironment.cs

```diff
@@ -13,13 +13,12 @@ public static string GetFolderPath(NuGetFolderPath folder)
             switch (folder)
             {
                 case NuGetFolderPath.MachineWideSettingsBaseDirectory:
-                    return Path.Combine(
-                        GetFolderPath(SpecialFolder.CommonApplicationData),
-                        "NuGet");
+                    return GetFolderPath(SpecialFolder.CommonApplicationData);
 
                 case NuGetFolderPath.MachineWideConfigDirectory:
                     return Path.Combine(
                         GetFolderPath(NuGetFolderPath.MachineWideSettingsBaseDirectory),
+                        "nuget",
                         "Config");
 
```

Comment:
actually machine wide is only used on windows now, we don't load machine wide setting on other platforms, this is a know bug I will fix later. so it will not cause casting issue.